lang/cs/Org.Apache.REEF.Common.Tests/Telemetry/MetricsTests.cs

```diff
@@ -0,0 +1,135 @@
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Common.Telemetry;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Xunit;
+
+namespace Org.Apache.REEF.Common.Tests.Telemetry
+{
+    public class MetricsTests
+    {
+        /// <summary>
+        /// Test IMetrics, ICounters and IEvaluatorMetrics API.
+        /// </summary>
+        [Fact]
+        public void TestEvaluatorMetricsCountersOnly()
+        {
+            var evalMetrics = TangFactory.GetTang().NewInjector().GetInstance<IEvaluatorMetrics>();
+            var metrics = evalMetrics.GetMetrics();
+            metrics.TryRegisterMetric(new Counter("counter1", "counter1 description"));
+            metrics.TryRegisterMetric(new Counter("counter2", "counter2 description"));
+            ValidateMetric(metrics, "counter1", MetricType.Counter, 0);
+            ValidateMetric(metrics, "counter2", MetricType.Counter, 0);
+
+            metrics.IncrementCounter("counter1", 3);
+            metrics.IncrementCounter("counter1", 1);
+            metrics.IncrementCounter("counter2", 2);
+            metrics.IncrementCounter("counter2", 3);
+            ValidateMetric(metrics, "counter1", MetricType.Counter, 4);
+            ValidateMetric(metrics, "counter2", MetricType.Counter, 5);
+            var counterStr = evalMetrics.Serialize();
+
+            var metrics2 = new EvaluatorMetrics(counterStr);
+            var counters2 = metrics2.GetMetrics();
+            ValidateMetric(counters2, "counter1", MetricType.Counter, 4);
+            ValidateMetric(counters2, "counter2", MetricType.Counter, 5);
+        }
+
+        [Fact]
+        public void TestEvaluatorMetricsMixed()
+        {
+            var evalMetrics = TangFactory.GetTang().NewInjector().GetInstance<IEvaluatorMetrics>();
+            var metrics = evalMetrics.GetMetrics();
+            metrics.TryRegisterMetric(new Counter("counter", "counter description"));
+            metrics.TryRegisterMetric(new DoubleGauge("double", "double description", DateTime.Now.Ticks, 3.14));
+
+            ValidateMetric(metrics, "counter", MetricType.Counter, 0);
+            ValidateMetric(metrics, "double", MetricType.Double, 3.14);
+
+            metrics.IncrementCounter("counter", 3);
+            metrics.IncrementCounter("counter", 1);
+            ValidateMetric(metrics, "counter", MetricType.Counter, 4);
+            metrics.TryGetValue("double", out MetricBase doubleMetric);
+            doubleMetric.Value = 4.13;
+            ValidateMetric(metrics, "double", MetricType.Double, 4.13);
+
+            var metricsStr = evalMetrics.Serialize();
+            var evalMetrics2 = new EvaluatorMetrics(metricsStr);
+            var metrics2 = evalMetrics2.GetMetrics();
+            ValidateMetric(metrics2, "counter", MetricType.Counter, Convert.ToInt32(4));
+            ValidateMetric(metrics2, "double", MetricType.Double, 4.13);
```

Comment:
We need a test that validates a metric as a series of values, i.e. [3.14, 4.13]?